tracing/src/test/java/com/palantir/tracing/TracersTest.java

```diff
@@ -257,6 +264,120 @@ public void testWrapRunnable_startsNewSpan() throws Exception {
         assertThat(Tracer.completeSpan().get().getOperation()).isEqualTo("outside");
     }
 
+    @Test
+    public void testWrapFuture_immediate() {
+        List<Span> observed = new ArrayList<>();
+        Tracer.subscribe("futureTest", observed::add);
+        try {
+            String operationName = "testOperation";
+            ListenableFuture<String> traced =
+                    Tracers.wrapListenableFuture(operationName, () -> Futures.immediateFuture("result"));
+            assertThat(traced).isDone();
+            assertThat(observed).hasSize(2);
+            // Inner operation must complete first to avoid confusion
+            assertThat(observed.get(0))
+                    .extracting(Span::getOperation)
+                    .isEqualTo(operationName + " initial");
+            assertThat(observed.get(1))
+                    .extracting(Span::getOperation)
+                    .isEqualTo(operationName);
+            assertThat(observed)
+                    .allSatisfy(span -> assertThat(span)
+                            .extracting(Span::getTraceId)
+                            .isEqualTo("defaultTraceId"));
+        } finally {
+            Tracer.unsubscribe("futureTest");
+        }
+    }
```

Comment:
If we'd fully converted this repo to JUnit5 then I would have loved to use the @TestTracing things for these!